public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -103,6 +102,10 @@ export class PanelChrome extends PureComponent<Props, State> {
   canEditDashboard = () => Boolean(this.props.dashboard.meta.canEdit || this.props.dashboard.meta.canMakeEditable);
 
   canAddAnnotation = () => {
+    if (config.isPublicDashboardView) {
```

Comment:
You can remove this, it's not needed. the check further down is using this.canEditDashboard() checks this.props.dashboard.meta.canEdit which will be false for public dashboard